interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,376 @@
+            SearchResults.clear();
+
+            $scope.searchGenes = function() {
+              if (!$scope.genesToAdd) { //if the query is empty
+                return false;
+              }
+              $scope.loadingSearchResults = true;
+
+              $rootScope.$broadcast( 'results.loadingSearchResults' );
+              qparams = {'query': $scope.genesToAdd.query};
+
+              if ($scope.organism) {
+                qparams['organism'] = $scope.organism;
+              }
+
+              SearchResults.search(qparams);
+            };
+
+            $scope.$on('results.searchResultsReturned', function() {
+              $scope.loadingSearchResults = false;
+            });
+
+          }
+        ],
+        replace: true,
+        restrict: "E",
+        scope: {
+            query: '@'
+        },
+        templateUrl: 'gene/gene-search-form.tpl.html'
+    };
+}])
+
+// Directive for table containing search results
+.directive('searchResultTable', ['SearchResults', function( SearchResults ) {
+  return {
+    controller: ['$scope', 'SearchResults', function($scope, SearchResults) {
+      $scope.currentPage = 1;
+      $scope.itemsPerPage = 10;
+      $scope.totalResults = 0;
+      $scope.maxSize = 10;
+      $scope.resultsForPage = [];
+      $scope.searchResults = SearchResults.getSearchResults();
+      $scope.loadingSearchResults = false;
+
+      $scope.addAllNonAmbiguous = function() {
+      // Function to automatically add all genes that only have one search result
+        var searchResults = SearchResults.getSearchResults();
+        for (var key in searchResults) {
+          var results = searchResults[key];
+          if (results['found'].length <= 1) {
+            if (results['found'][0]) {
+              var gene = results['found'][0];
+              SearchResults.remove(key);
+            }
+          }
+        }
+      };
+
+      $scope.removeNotFound = function() {
+      // Function to get rid of all the queries that returned no results.
+          var searchResults = SearchResults.getSearchResults();
+          for (var key in searchResults) {
+            var results = searchResults[key];
+            if (results['found'].length === 0) {
+              SearchResults.remove(key);
+            }
+          }
+      };
+    }],
+
+    link: function(scope, element, attr) {
+      scope.$on('results.update', function() {
+        scope.totalResults = SearchResults.size();
+        var begin = ((scope.currentPage-1)*scope.itemsPerPage);
+        var end = begin + scope.itemsPerPage;
+        scope.resultsForPage = SearchResults.getQueries().slice(begin, end);
+      });
+
+      scope.$on('results.loadingSearchResults', function() {
+        scope.loadingSearchResults = true;
+      });
+
+      scope.$on('results.searchResultsReturned', function() {
+        scope.loadingSearchResults = false;
+      });
+
+      //Watch for page changes and update
+      scope.$watch('currentPage', function() {
```

Comment:
Lines 203-205 seem identical to lines 188-190. If you want, you can define a separate function to reduce the redundancy. (since only three extra lines are involved, it is up to you.)